sql/api/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -476,8 +478,8 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
    * 4. Otherwise, `this` and `that` are considered as conflicting schemas and an exception would be
    *    thrown.
    */
-  private[sql] def merge(that: StructType): StructType =
-    StructType.merge(this, that).asInstanceOf[StructType]
+  private[sql] def merge(that: StructType, caseSensitive: Boolean = true): StructType =
```

Comment:
`spark.sql.caseSensitive` is false by default, maybe we should have false as default here?

Reply:
I'm being conservative here as the previous code is case-sensitive.